plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.block_theme.side_effects.spec.ts

```diff
@@ -1065,4 +1065,51 @@ test.describe( 'Product Collection', () => {
 			);
 		} );
 	} );
+
+	test.describe( 'Preview mode in generic archive templates', () => {
+		const genericArchiveTemplates = [
+			{
+				name: 'Products by Tag',
+				path: 'woocommerce/woocommerce//taxonomy-product_tag',
+			},
+			{
+				name: 'Products by Category',
+				path: 'woocommerce/woocommerce//taxonomy-product_cat',
+			},
+			{
+				name: 'Products by Attribute',
+				path: 'woocommerce/woocommerce//taxonomy-product_attribute',
+			},
+		];
+
+		genericArchiveTemplates.forEach( ( { name, path } ) => {
+			test( `${ name } template`, async ( { page, pageObject } ) => {
+				await pageObject.replaceProductsWithProductCollectionInTemplate(
+					path
+				);
+
+				const editorFrame = page.frameLocator(
+					'iframe[name="editor-canvas"]'
+				);
+				const previewButtonLocator = editorFrame.locator(
+					'button.wc-block-product-collection__preview-button:has-text("Preview")'
```

Comment:
We really need to avoid such class locators in our code. Changing a class should not break the test, it will push maintenance burden to the next person.

I would suggest using role combined with has-text here.

Reply:
Made the changed in [8c9b9d7](https://github.com/woocommerce/woocommerce/pull/46369/commits/8c9b9d765abe0b727e09d12f539f1bc7d55bb97a)